### PR TITLE
`lg` cli

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
## ✍️ Proposed changes

Creates a single package (`@lg-tools/cli`) that builds a Command Line interface. 
This package is intended to be shared across other LeafyGreen repositories, as well as used in projects that use LeafyGreen UI components.

Current supported subcommands include: 

```
lg <subcommand>
  create [options] <name>                 Creates a new package with the provided name
  install [options] [packages...]         Installs LeafyGreen packages to the current app
  link [options] <destination>            Link local LeafyGreen packages to a destination app.
  unlink [options] <destination>          Unlink local LeafyGreen packages from a destination app.
  lint [options]                          Checks code and package.json formatting
  test [options]                          Tests leafygreen-ui packages with unified config.
  update|upgrade [options] [packages...]  Updates installed LeafyGreen packages based on the specified range
  validate [options]                      Validates build integrity, and package.json dependencies
  build-package                           Builds a package
  build-ts                                Builds a package's TypeScript definitions
  build-tsdoc                             Builds a package's TSDoc file
  help [command]                          display help for command
``` 

### Architecture

Inside `tools/cli` we use Commander to create the `lg` command, and all subcommands. 

`cli` depends on several other `@lg-tools` packages. When a subcommand is called, we fire an action function defined in one of the other `@lg-tools/` dependencies.

We define a `bin` property in `cli/package.json` which declares what script to run when we call `> lg` from the terminal

### CODE REVIEW

File: .github/workflows/pr.yml [hunk 1]
```diff
@@ -22,7 +22,7 @@ jobs:
           path: |
             **/dist
             !**/node_modules
-          key: ${{ runner.os }}-build-cache-${{ hashFiles('**/rollup.config.js', '**/babel.config.js', '**/src/', '**/scripts/', '!**/node_modules', '!**/dist') }}
+          key: ${{ runner.os }}-build-cache-${{ hashFiles('**/src/', '!**/node_modules', '!**/dist') }}
```
Comment (bruugey) [inline .github/workflows/pr.yml:25]: what's the expected gain here

File: package.json [hunk 2]
```diff
@@ -9,35 +9,31 @@
     "npm": ">= 8.19.4"
   },
   "scripts": {
+    "lg": "lg",
```
Comment (bruugey) [inline package.json:12]: do we need to define this here, or is it a global install on your OS? (not opposed, just curious!)

File: packages/badge/package.json [hunk 3]
```diff
@@ -37,5 +38,8 @@
   },
   "bugs": {
     "url": "https://jira.mongodb.org/projects/PD/summary"
+  },
+  "devDependencies": {
+    "@lg-tools/cli": "^0.0.1"
```
Comment (bruugey) [inline packages/badge/package.json:43]: this can't be inherited from the larger workspace?

File: package.json (hunk 2 above)
Comment (TheSonOfThomp) [reply package.json:12]: No ,I just had this here for testing

File: packages/badge/package.json (hunk 3 above)
Comment (bruugey) [reply packages/badge/package.json:43]: or `prepare`? but that's fine with me

File: tools/link/README.md [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+Link and Unlink packages to a consuming repository
+
+## Usage
+
+Use from `@lg-tools/cli`.
+
+### Link
+
+Run `link` from the leafygreen repository to link packages to a consuming app.
```
Comment (bruugey) [inline tools/link/README.md:11]: does this work from all of our repos?

Comment (TheSonOfThomp) [reply tools/link/README.md:11]: Yep! It'll link all workspaces defined in `lg.config`

File: packages/a11y/package.json [hunk 5]
```diff
@@ -32,5 +33,7 @@
   },
   "bugs": {
     "url": "https://jira.mongodb.org/projects/PD/summary"
-  }
-}
\ No newline at end of file
+  },
+  "devDependencies": {
+   }
```
Comment (spark33) [inline packages/a11y/package.json:38]: Empty object

File: packages/toast/src/ToastContext/ToastReducer/useToastReducer.spec.ts [hunk 6]
```diff
@@ -1,4 +1,4 @@
-import { act } from 'react-test-renderer';
```
Comment (spark33) [inline packages/toast/src/ToastContext/ToastReducer/useToastReducer.spec.ts:1]: if this is a change that needs to be made for things to work, is this going to cause issues when merging with react 18?

File: tools/create/src/templates/component/src/Component/component.index.template.ts [hunk 7]
```diff
@@ -0,0 +1,8 @@
+import { TemplateParameters } from '../../../../create.types';
+
+export const componentIndex = ({
+  packageNamePascal,
+}: Pick<TemplateParameters, 'packageNamePascal'>) => `
+export  { ${packageNamePascal} } from './${packageNamePascal}';
+export { ${packageNamePascal}Props } from './${packageNamePascal}.types';
+`;
```
Comment (spark33) [inline tools/create/src/templates/component/src/Component/component.index.template.ts:8]: 🚀

File: tools/lint/src/npmPkgJsonLint.ts [hunk 8]
```diff
@@ -1,29 +1,39 @@
...
-const npmPackageJsonLinter = new NpmPackageJsonLint({
-  cwd: rootDir,
-  patterns: ['**/package.json'],
-  configFile: npmPkgLintConfigPath,
-});
+// const npmPackageJsonLinter = new NpmPackageJsonLint({
+//   cwd: rootDir,
+//   patterns: ['**/package.json'],
+//   configFile: npmPkgLintConfigPath,
+// });
```
Comment (spark33) [inline tools/lint/src/npmPkgJsonLint.ts:18]: commented code!

File: packages/toast/src/ToastContext/ToastReducer/useToastReducer.spec.ts (hunk 6 above)
Comment (TheSonOfThomp) [reply packages/toast/src/ToastContext/ToastReducer/useToastReducer.spec.ts:1]: Definitely needs to be made. I don't think it should cause issues with React 18 (but we'll cross that bridge later)

**Pull request merged**